lib/java/src/org/apache/thrift/server/TThreadPoolServer.java

```diff
@@ -303,6 +303,10 @@ public void run() {
       if (outputTransport != null) {
         outputTransport.close();
       }
+
+      if (client_.isOpen()) {
```

Comment:
This was also addressed in THRIFT-3083 where the Simple, Threaded, and ThreadPool server serve() routine was consolidated; some servers closed the client and some did not.